apps/browser/app/api/goal/generate-title/route.ts

```diff
@@ -0,0 +1,73 @@
+import { getAuthOptions } from "@/lib/api/authOptions";
+import { createOpenAIApiClient } from "@/lib/api/utils/openai";
+import { createSupabaseClient } from "@/lib/supabase/createSupabaseClient";
+import { getServerSession } from "next-auth";
+import { NextRequest, NextResponse } from "next/server";
+
+export async function GET(request: NextRequest) {
+  const searchParams = request.nextUrl.searchParams;
+  const chatId = searchParams.get("chatId");
+  const prompt = searchParams.get("prompt");
+
+  if (!chatId || !prompt) {
+    return NextResponse.json({}, { status: 400 });
+  }
+
+  // Make sure user is authenticated
+  const session = await getServerSession(getAuthOptions());
+  const email = session?.user?.email;
+  if (!session || !email) {
+    return NextResponse.json({}, { status: 401 });
+  }
+  const supabase = createSupabaseClient(session.supabaseAccessToken as string);
+  // Only can only fetch chats associated with its ID, so if it doesn't return
+  // anything it means it is querying a chat from other user
```

Comment:
Not sure this comment is necessary, the query is pretty self explanatory

Reply:
i don't think is that self explanatory as it looks like the user may query any chat id - but i will remove it anyways